### PR TITLE
Add framework plugin

### PR DESCRIPTION
**Pull request by andreaspeters**
As I mentioned, I add a framework plugin to mesos-cli and some new features for the already existing task plugin.

The features are:

- show all framework
- inspect frameworks and tasks
- disable the limitation of tasks in the task list

### CODE REVIEW

**Comment by cf-natali** [comment]
Hey,

I'm on holiday so I'll have a look when I'm back, unless someone else beats me to it.

**Comment by andreaspeters** [comment]
:joy: Take your time and enjoy your holiday.

File: src/python/cli_new/lib/cli/http.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def read_endpoint(addr, endpoint, config, query=None):
 
     try:
         addr = cli.util.sanitize_address(addr)
+
```
Comment (cf-natali) [inline src/python/cli_new/lib/cli/http.py:40]: Nit: why the newline?

File: src/python/cli_new/lib/cli/http.py [hunk 2]
```diff
@@ -64,7 +65,7 @@ def read_endpoint(addr, endpoint, config, query=None):
                            .format(url=url, error=str(exception)))
 
 
-def get_json(addr, endpoint, config, condition=None, query=None):
+def get_json(addr, endpoint, config, query=None):
```
Comment (cf-natali) [inline src/python/cli_new/lib/cli/http.py:68]: Are you sure that nothing uses `condition`?

Also, if you remove t, please remove the mention in the docstring as well:
```
subject to the condition specified in 'condition'. If we are
```

**Review by cf-natali — COMMENTED** [review]
I think it would be better to split the changes to the task plugin into its own MR.

**Comment by andreaspeters** [comment]
> I think it would be better to split the changes to the task plugin into its own MR.

Oh please don't push me to split it. :-( Thats very painfull. There will be more PR's for the mesos-cli in the future. Then I will keep an eye to separate it.

File: src/python/cli_new/lib/cli/http.py (hunk 1 above)
Comment (andreaspeters) [reply src/python/cli_new/lib/cli/http.py:40]: No reason. :man_shrugging:  will remove it.

File: src/python/cli_new/lib/cli/http.py (hunk 2 above)
Comment (andreaspeters) [reply src/python/cli_new/lib/cli/http.py:68]: Not the mesos_cli. I had a look. But of course, I will also feel better u you keep an eye on it. :-)

**Comment by cf-natali** [comment]
> > I think it would be better to split the changes to the task plugin into its own MR.
> 
> Oh please don't push me to split it. :-( Thats very painfull. There will be more PR's for the mesos-cli in the future. Then I will keep an eye to separate it.

Really?
The problem is that fundamentally this MR contains two unrelated changes, so it'd be nice to separate.
Either to review, commit, and revert if necessary...

**Comment by andreaspeters** [comment]
> > > I think it would be better to split the changes to the task plugin into its own MR.
> > 
> > 
> > Oh please don't push me to split it. :-( Thats very painfull. There will be more PR's for the mesos-cli in the future. Then I will keep an eye to separate it.
> 
> Really?
> The problem is that fundamentally this MR contains two unrelated changes, so it'd be nice to separate.
> Either to review, commit, and revert if necessary...

Charles u are killing me. :joy:  But I will do it.

**Comment by andreaspeters** [comment]
Shell I open three PR's? Technically  there are three kind of changes.

1. Add Framework Plugins
2. Update Task Plugins
3. Update mesos_cli lib.

What do u think?

**Comment by cf-natali** [comment]
Sounds good!

**Comment by cf-natali** [comment]
@andreaspeters how's the refactoring going? :)

**Comment by andreaspeters** [comment]
> @andreaspeters how's the refactoring going? :)

Didn't found the time the last week. Hopefully the next two weeks.

**Comment by cf-natali** [comment]
@andreaspeters I'm closing this one since you've split it into separate MRs.

**Pull request closed without merging**